cciskel-duffy

```diff
@@ -45,7 +46,109 @@ def print_classes(classes):
             for machine in clsdata:
                 print("  {}".format(machine))
 
-def job_take(args, ssids, allocated_class):
+
+def count_requested_instances(topo_data):
+    count = 0
+
+    for td in topo_data:
+        count += int(td['count'])
+
+    return count
+
+
+def write_inventory(args, allocated_class, topo_data=None):
+
+    a_ssid = None
+    first_machine_allocated = None
+
+    for machine in allocated_class:
+        if machine['job'] != '':
+            continue
+        if a_ssid is None:
+            a_ssid = machine['ssid']
+            first_machine_allocated = machine['name']
+        machine['job'] = args.jobid
+
+    inventory = {}
+
+    if topo_data:
+        n_allocated = 0
+        node_types = len(topo_data)
+
+        for nt in range(node_types):
+
+            nt_hosts = int(topo_data[nt]['count'])
+            metadata = topo_data[nt]['metadata']
+            groups = metadata['ansible-group']
+
+            for group in groups:
+                for k,v in metadata.iteritems():
+                    if k != 'ansible-group':
+                        gvs = '{}:vars'.format(group)
+                        if inventory.has_key(gvs):
+                            inventory[gvs][k] = v
+                        else:
+                            inventory[gvs] = {k: v}
+
+            for nt in range(nt_hosts):
+                host = allocated_class[n_allocated]
+
+                for group in groups:
+                    if inventory.has_key(group):
+                        if not host['name'] in inventory[group]:
+                            inventory[group].append(host['name'])
+                    else:
+                        inventory[group] = [host['name']]
+
+                n_allocated += 1
+                if n_allocated == args.count:
+                    break
+
+    assert n_allocated == args.count
+
+    with open('inventory.{}'.format(args.jobid), 'a+') as f:
+        # if args.topology exists, use it instead of args.ansible_group
+        if inventory:
+            for k, v in inventory.iteritems():
+                f.write('[{}]\n'.format(k))
+                try:
+                    for q,w in v.iteritems():
+                        f.write("{}='{}'\n".format(q,w))
+                except Exception as e:
```

Comment:
Hm...is this basically dispatching on whether the member is a list or a map?  Would feel a bit cleaner as:

```
if isinstance(v, dict):
  for q,w in v.iteritems():
    f.write("{}='{}'\n".format(q,w))
else:
  for items in v:
    f.write('{}\n'.format(items))
```